server/plugin.go

```diff
@@ -20,9 +23,82 @@ type Plugin struct {
 	configuration *configuration
 }
 
-// ServeHTTP demonstrates a plugin that handles HTTP requests by greeting the world.
+type apiResponse struct {
+	CookieName  string `json:"cookieName"`
+	CookieValue string `json:"cookieValue"`
+	StatusCode  int    `json:"statusCode"`
+}
+
+// ServeHTTP handles HTTP requests.
 func (p *Plugin) ServeHTTP(c *plugin.Context, w http.ResponseWriter, r *http.Request) {
-	fmt.Fprint(w, "Hello, world!")
+	switch path := r.URL.Path; path {
+	case "/api/v1/check":
+		p.handleCookieCheck(w, r)
+	}
 }
 
-// See https://developers.mattermost.com/extend/plugins/server/reference/
+// handleCookieCheck function is used to handle the checks for the existense of the Canary cookie.
+func (p *Plugin) handleCookieCheck(w http.ResponseWriter, r *http.Request) {
+	userID := r.Header.Get("Mattermost-User-ID")
+	if userID == "" {
+		http.Error(w, "Not authorized", http.StatusUnauthorized)
+		return
+	}
+	cookie, err := r.Cookie("canary")
+	if err != nil {
+		p.API.LogDebug("Canary cookie does not exist for user " + userID)
+		err := p.canaryPercentage(w, userID)
+		if err != nil {
+			p.API.LogError("Unable to set cookie" + err.Error())
+			http.Error(w, "Unable to set cookie", http.StatusInternalServerError)
+			return
+		}
+	} else if cookie.Value == "never" {
+		p.API.LogDebug("Canary cookie is set to never for user " + userID)
+		err := p.canaryPercentage(w, userID)
+		if err != nil {
+			p.API.LogError("Unable to set cookie" + err.Error())
+			http.Error(w, "Unable to set cookie", http.StatusInternalServerError)
+			return
+		}
+	}
+}
+
+// canaryPercentage function is used to check the canary percentage and call the addCookie function.
+func (p *Plugin) canaryPercentage(w http.ResponseWriter, userID string) error {
```

Comment:
s/canaryPercentage/addCanaryCookieBasedOnPercentage/

Just to be a little more clear about what the method is doing

Reply:
Done ;)